src/java.base/share/classes/java/security/cert/CertPath.java

```diff
@@ -177,18 +177,19 @@ public String getType() {
      * @return true if the specified object is equal to this certification path,
      * false otherwise
      */
+    @Override
     public boolean equals(Object other) {
         if (this == other)
             return true;
 
         return other instanceof CertPath that
-                && that.getType().equals(this.type)
+                && this.type.equals(that.getType())
                 && this.getCertificates().equals(that.getCertificates());
     }
 
     /**
-     * Returns the hashcode for this certification path. The hash code of
-     * a certification path is defined to be the result of the following
+     * {@return the hashcode value for this certification path} The hash code
```

Comment:
This looks a bit odd w/o a period at the end of the `}`. Is it optional? [The javadoc spec](https://docs.oracle.com/en/java/javase/20/docs/specs/javadoc/doc-comment-spec.html#return) isn't very clear on that - perhaps you could clarify that in the spec.

If it is optional, my general preference would be to include the period, as it makes the code more readable.

Reply:
While I agree it looks "a bit odd", the problem is having to address the two ways in which the tag is rendered:

1. As the first sentence. In this case, the text in the braces is preceded by `Returns` and followed by a period.
2. As the equivalent for an `@return` tag, where `Returns:` is a bold heading, and the content is a phrase, and (generally) not followed by a period.

The current guideline is that the period should not be present. Elsewhere, the "oddness" is reduced by putting
a newline after the tag, so that the `{@return}` tag stands on a line by itself.  To forestall any possible suggestion,
I would not be in favor of looking outside the tag to see if there is a period following the tag.

We can clarify the spec.

If you are uncomfortable with the usage here, it is always possible to avoid using the short inline form, and to "duplicate" the text and use an old-fashioned plain first sentence, and an old-fashioned `@return` "block" tag